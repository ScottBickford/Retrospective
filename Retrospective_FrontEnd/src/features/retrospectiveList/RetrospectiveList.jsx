import React, { Fragment } from 'react';
import { format } from 'date-fns';

const RetrospectiveList = ({ data }) => {
  return (
    <Fragment>
      {data.list.length === 0 && <div>There are no items</div>}
      {data.list.length > 0 && (
        <table border={1} cellSpacing={0} cellPadding={4}>
          <thead>
            <tr>
              <td>Name</td>
              <td>Summary</td>
              <td>Date</td>
              <td>Participants</td>
              <td></td>
            </tr>
          </thead>
          <tbody>
            {data.list.map(item => (
              <tr key={item.name}>
                <td>{item.name}</td>
                <td>{item.summary}</td>
                <td>{format(new Date(item.date), 'd MMMM yyyy')}</td>
                <td>
                  {item.participants &&
                    item.participants.map(participant => {
                      return <div key={participant}>{participant}</div>;
                    })}
                </td>
                <td><a href={`/view/${item.retrospectiveId}`}>View</a></td>
              </tr>
            ))}
          </tbody>
        </table>
      )}
    </Fragment>
  );
};

export default RetrospectiveList;
