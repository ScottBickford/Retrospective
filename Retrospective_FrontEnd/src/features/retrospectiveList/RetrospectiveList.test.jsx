import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RetrospectiveList from './RetrospectiveList';

const data = {list: [{name: 'test', summary: 'test', date: '2012-02-02', participants: ['Person 1', 'Person 2']}]};
describe('RetrospectiveList', () => {
  test('snapshot renders', () => {    
    const component = renderer.create(<RetrospectiveList data={data} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Input', () => {
    const { getByText } = render(<RetrospectiveList data={data} />);
    const buttonElement = getByText(/Person 1/i);
    expect(buttonElement).toBeInTheDocument();
  });
});