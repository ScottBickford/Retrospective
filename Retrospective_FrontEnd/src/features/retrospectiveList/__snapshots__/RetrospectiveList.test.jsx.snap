// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrospectiveList snapshot renders 1`] = `
<table
  border={1}
  cellPadding={4}
  cellSpacing={0}
>
  <thead>
    <tr>
      <td>
        Name
      </td>
      <td>
        Summary
      </td>
      <td>
        Date
      </td>
      <td>
        Participants
      </td>
      <td />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        test
      </td>
      <td>
        test
      </td>
      <td>
        2 February 2012
      </td>
      <td>
        <div>
          Person 1
        </div>
        <div>
          Person 2
        </div>
      </td>
      <td>
        <a
          href="/view/undefined"
        >
          View
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
